segmentstore/storage/src/main/java/io/pravega/segmentstore/storage/chunklayer/SystemJournal.java

```diff
@@ -1573,6 +1576,85 @@ private <R> CompletableFuture<R> executeExclusive(Callable<CompletableFuture<R>>
+                            .thenApplyAsync( readBackInfo -> {
+                                if (readBackInfo.getEpoch() > epochInfo.getEpoch()) {
+                                    throw new CompletionException(
+                                            new IllegalStateException(String.format("Unexpected epochInfo. Expected = {} actual = {}", epochInfo, readBackInfo)));
+                                }
+                                if (!epochInfo.equals(readBackInfo)) {
+                                    throw new CompletionException(
+                                            new IllegalStateException(String.format("Unexpected epochInfo. Expected = {} actual = {}", epochInfo, readBackInfo)));
+                                }
+                                return null;
+                            }, executor)
+                            .handleAsync((v, e) -> {
+                                if (null != e) {
+                                    val ex = Exceptions.unwrap(e);
+                                    if (ex instanceof IllegalStateException) {
+                                        log.warn("{}: Error while saving epoch info to file {}. info = {}", containerId, chunk, epochInfo, e);
+                                        throw new CompletionException(e);
+                                    }
+                                    if (attempts.incrementAndGet() > MAX_FLUSH_ATTEMPTS) {
+                                        log.warn("{}: All attempts exhausted while saving epoch to File = {}. info = {}", containerId, chunk, epochInfo, e);
+                                        throw new CompletionException(e);
+                                    }
+                                    log.warn("{}: Error while saving epoch info to LTS. File = {}. info = {}", containerId, chunk, epochInfo, e);
+                                } else {
+                                    // No exception we are done
+                                    log.info("{}: Epoch info saved successfully. info = {}", containerId, chunk);
+                                    isDone.set(true);
+                                }
+                                return null;
+                            }, executor),
+                    this.executor);
+        } catch (IOException e) {
+            return CompletableFuture.failedFuture(new ChunkStorageException(chunk, "Unable to serialize", e));
+        }
+    }
+
+    private CompletableFuture<EpochInfo> readEpochInfo(String chunk, ChunkedSegmentStorage chunkedSegmentStorage, int readLength) {
+        val readAtOffset = new AtomicLong(0);
+        val readBuffer = new byte[readLength];
+        return chunkedSegmentStorage.getChunkStorage().read(ChunkHandle.readHandle(chunk), readAtOffset.get(), readLength, readBuffer, 0)
+                .thenApplyAsync( v -> {
```

Comment:
reference to chunkedSegmentStorage not needed.
use this.getChunkStorage() instead.

Reply:
done.